src/main/java/net/minecraftforge/common/crafting/CraftingHelper.java

```diff
@@ -224,7 +224,7 @@ public static ItemStack getItemStack(JsonObject json, JsonContext context)
                 if(element.isJsonObject())
                     nbt = JsonToNBT.getTagFromJson(GSON.toJson(element));
                 else
-                    nbt = JsonToNBT.getTagFromJson(element.getAsString());
+                    nbt = JsonToNBT.getTagFromJson(JsonUtils.getString(element, "nbt"));
```

Comment:
This is not correct, `element` is already the value of "nbt". I think you meant to pass `nbt` here?

Reply:
See the implementation of `JsonUtils.getString(JsonElement, String)`. This is not the `JsonObject` version - it's taking the element name for error reporting only.

Reply:
That's annoyingly unclear, but alright.